### PR TITLE
Bump CAPI to v1.5.3

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind support

**What this PR does / why we need it**:
This PR bumps CAPI to v1.5.3

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [x] adds or updates e2e tests

**Release note**:

```release-note
Bump CAPI to v1.5.3
```

### CODE REVIEW

File: test/e2e/data/e2e_conf.yaml [hunk 1]
```diff
@@ -116,8 +116,8 @@ providers:
   - name: aws
     type: InfrastructureProvider
     versions:
-      - name: v1.5.2 # latest published release in the v1beta1 series; this is used for v1beta1 --> v1beta2 clusterctl upgrades test only.
-        value: "https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v1.5.2/infrastructure-components.yaml"
+      - name: v1.5.3 # latest published release in the v1beta1 series; this is used for v1beta1 --> v1beta2 clusterctl upgrades test only.
+        value: "https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/download/v1.5.3/infrastructure-components.yaml"
```
Comment (AndiDog) [inline test/e2e/data/e2e_conf.yaml:120]: This shouldn't be changed – it's the CAPA version, not the CAPI version (see test failure)

Comment (Ankitasw) [reply test/e2e/data/e2e_conf.yaml:120]: nice catch

**Comment by Ankitasw** [comment]
PR needs review.

**Pull request merged**